### ISSUE TITLE
Track gradients and activations of models

### ISSUE THREAD

**Issue by f-hafner** [odissei-lifecourse/life-sequencing-dutch#47]
This should happen at the start -- to check initialization -- , and ideally during regular checkpointing

- [ ] log gradients & activations. with lightining? https://www.reddit.com/r/deeplearning/comments/152ukk5/best_practice_for_saving_logitsactivation_values/
- [ ] use tensorboard logger
- [ ] open the tensorboard files in a python environment on the RA? the following works
   `M:\mambaforge\PROJNR\Scripts\jupyter-notebook.exe --notebook-dir="H:/"`
- [ ] need to ask CBS to install the environment.

First, try this workflow with some dummy model trained on snellius.

**Comment by f-hafner** [comment]
probably we should have this option also for the graph models

**Comment by tanzir5** [comment]
This is related to #145

**Comment by tanzir5** [comment]
We can log the gradient min, median, max, and L2 norm of sum of gradients, for every 50th batch or something like that. We get to know if gradient explosion or vanishing happens.

We can also log the output embedding's min, median, max as well as the standard deviation of the output embedding matrix of size (batch_size x seq_len x dim) [ignoring padding token] for every 50th batch. Here we get to know if the embeddings are similar or dissimilar using the std. deviation.

**Comment by f-hafner** [comment]
closing as not planned
